src/gui/editors/PianoRoll.cpp

```diff
@@ -1816,42 +1778,52 @@ void PianoRoll::mousePressEvent(QMouseEvent * me )
 				m_mouseDownKey = m_startKey;
 				m_mouseDownTick = m_currentPosition;
 
+				bool resize = isHoveringResize(current_note);
+
+				m_action = resize ? ActionResizeNote : ActionMoveNote;
+
 				//If clicked on an unselected note, remove selection and select that new note
 				if (!m_currentNote->selected())
 				{
 					clearSelectedNotes();
 					m_currentNote->setSelected( true );
+
+					// Bulldozer on unselected note selects all notes to the right
+					if (m_editMode == ModeBulldozer && !resize && !shift && !created_new_note)
+					{
+						m_action = ActionBulldozerMove;
+						for (Note* note: m_midiClip->notes())
```

Comment:
If we do `const auto&` here it will evaluate to `Note* const &`, which doesn't prevent changes to the note. I'd guess that would be the biggest intention of using `const` here, since it's always an lvalue (at least for now).

If I want to protect the note, do you think I should go with `const auto*` or `const Note*` ?

Fot the other for-loops where the note is modified I'm just gonna go with `auto&` since *if* we move away from pointers in the future it will evaluate to `const Note&` and fail to build.

Reply:
> If I want to protect the note, do you think I should go with `const auto*` or `const Note*` ?

There was a discussion about this in Discord. From my perspective, there truly is no right or wrong answer if you really want to protect the actual ``Note*`` itself. I wish we didn't have to make everything pointers because of Qt. With accessing these objects by value, ``const auto&`` would work really well here with no question. In that case, I'm still somewhat for ``const auto&``, because when ``Note*`` becomes a ``Note``, the ``Note`` will actually be immutable. Also, I find that ``const auto&`` just works in a lot of scenarios, even when the codebase changes. It's easier to reason with something immutable, the type is deduced correctly by the compiler, and it doesn't make copies.

> In the for-loops where the note is modified I'm just gonna go with auto& since if we move away from pointers in the future it will evaluate to const Note& and fail to build.

I agree. I think an import consideration at least minimize changes given we move away from using pointers and calling ``new`` where we don't need to. However, in my modernize-loop-convert PR, I just left it as ``const auto&`` rather than ``auto&`` where possible because generally, I believe immutability should be used where possible. So it truly is a conflicting situation.

If you use ``auto&`` in places where you are modifying ``Note`` through a ``Note*``, I'm not really against it though. I don't see a right or wrong answer.

Reply:
> If you use `auto&` in places where you are modifying `Note` through a `Note*`, I'm not really against it. I don't see a right or wrong answer.

In those cases the answer is simple: `const auto&` now doesn't provide any protection, and `const auto&` in the future will fail to compile, so no `const` in those places.

Don't worry about doing that in the modernize-loop-convert PR though, because if we move away from pointers we'll find all the places where the introduced const breaks things, and in all other places it will be fixed without anyone thinking about it, so that's nice. Plus we'd have to change all `->` anyway